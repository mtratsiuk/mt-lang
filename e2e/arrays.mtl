(def printArray
  (|arr|
    (print arr)))

(def printSomeArray
  (||
    (print [1 2 3])))

(printArray [
  "one"
  2
  3])

(printSomeArray)
