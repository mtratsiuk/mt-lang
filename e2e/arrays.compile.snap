const __mtl_curry = f => (...args) => {
  if (args.length >= f.length) {
    return f(...args);
  }

  return __mtl_curry(f.bind(null, ...args));
};

const printArray = ((arr) => {
  return console.log(String(arr));
});

const printSomeArray = (() => {
  return console.log(String([1, 2, 3]));
});

printArray(["one", 2, 3]);

printSomeArray();
